buildSrc/src/main/java/org/opensearch/gradle/zipplugin/ZipPublish.java

```diff
@@ -0,0 +1,122 @@
+    public final static String MAVEN_ZIP_PUBLISH_TASK = "publish"
+        + ZipPublishUtil.capitalize(PUBLICATION_NAME)
+        + "PublicationTo"
+        + ZipPublishUtil.capitalize(STAGING_REPO)
+        + "Repository";
+    public final static String MAVEN_ZIP_PUBLISH_POM_TASK = "generatePomFileFor"
+        + ZipPublishUtil.capitalize(PUBLICATION_NAME)
+        + "Publication";
+    public final static String LOCALMAVEN = "publishToMavenLocal";
+    public final static String LOCAL_STAGING_REPO_PATH = "/build/local-staging-repo";
+    public static String BUILD_DISTRIBUTIONS_LOCATION = "/build/distributions/";
+
+    private void configMaven() {
+        final Path buildDirectory = this.project.getRootDir().toPath();
+        this.project.getPluginManager().apply(MavenPublishPlugin.class);
+        this.project.getExtensions().configure(PublishingExtension.class, publishing -> {
+            publishing.repositories(repositories -> {
+                repositories.maven(maven -> {
+                    maven.setName(STAGING_REPO);
+                    maven.setUrl(buildDirectory.toString() + LOCAL_STAGING_REPO_PATH);
+                });
+            });
+            System.out.println("Starting " + MAVEN_ZIP_PUBLISH_TASK + " task");
+            publishing.publications(publications -> {
+                publications.create(PUBLICATION_NAME, MavenPublication.class, mavenZip -> {
+                    ZipPublishExtension extset = this.project.getExtensions().findByType(ZipPublishExtension.class);
+                    // Getting the Zip group from created extension
+                    String zipGroup = extset.getZipgroup();
+                    String zipArtifact = getProperty("zipArtifact");
+                    // Getting the Zip version from gradle property with/without added snapshot and qualifier
+                    String zipVersion = System.getProperty("opensearch.version");
+                    String version = null;
+                    String extraSuffix = null;
+                    if (zipVersion != null) {
+                        StringTokenizer st = new StringTokenizer(zipVersion);
+                        version = st.nextToken("-") + ".0";
+                        try {
+                            extraSuffix = zipVersion.substring(zipVersion.indexOf("-"));
+                        } catch (Exception e) {
+                            System.out.println("");
+                        }
+                    }
+                    String finalZipVersion = version + extraSuffix;
+                    String zipFilePath = null;
+                    // -PzipFilePath=/build/distributions/opensearch-job-scheduler-2.0.0.0-alpha1-SNAPSHOT.zip
+                    if (getProperty("zipFilePath") != null) {
+                        BUILD_DISTRIBUTIONS_LOCATION = getProperty("zipFilePath");
+                        zipFilePath = BUILD_DISTRIBUTIONS_LOCATION + zipArtifact + "-" + finalZipVersion + ".zip";
```

Comment:
This logic is the same, just `BUILD_DISTRIBUTIONS_LOCATION` changes.

All these CAPITALIZED variables are weird. Only CONSTANTS should be capitalized and I suspect half of them aren't needed anyway. Cleanup.

Reply:
I have considered them as constants (using variable modifiers static and final ) :) 
As these vars are constants, and have added final to them, these referenced one or multiple times throughout the code, also these should not be changed moving forward with any logic, hence considering these I have CAPITALIZED them.  Please correct me if my though process is wrong, tomorrow if we have to change some name setting, staging repo settings and distributions path, all we need is to just update the constants, no need to touch the actual methods.

Reply:
Agree with `BUILD_DISTRIBUTIONS_LOCATION`, also when I read back again, better it would be as `zipDistributionLocation`, fixed this in my latest commit.